src/main/java/org/jboss/tm/TransactionManagerEnvironment.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat Middleware LLC, and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.tm;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.transaction.UserTransaction;
+
+import org.jboss.tm.usertx.UserTransactionOperationsProvider;
+import org.jboss.tm.usertx.client.ServerVMClientUserTransactionOperationsProvider;
+
+/**
+ * <p>
+ * Class gathering settings for transaction manager when using SPI.
+ * <p>
+ * The main purpose is to provide way how Narayana get information
+ * what implementation of particular interface to use.
+ *
+ * @author Ondra Chaloupka <ochaloup@redhat.com>
+ */
+public final class TransactionManagerEnvironment {
```

Comment:
I think having registry could be possible but harder to implement. I'm again not sure if duplication of registry in spi and narayana is good idea. (if i'm not the same page with term 'registry' :)